include/gatekeeper_gk.h

```diff
@@ -99,17 +99,26 @@ struct neighbor_hash_table {
 	/* The tables that store the Ethernet headers. */
 	struct ether_cache *cache_tbl;
 
-	/* TODO Use a spin lock to edit the hash table. */
 	struct rte_hash *hash_table;
 };
 
 /* The gk forward information base (fib). */
 struct gk_fib {
 
+	/* Use a spin lock to edit the FIB table. */
+	rte_spinlock_t lock;
```

Comment:
(1) I think we can have a single lock for the FIB table, and carefully edit the FIB table to allow readers go without a lock. The idea is to somehow follow RCU. Notice that reader either goes through a prefix lookup or a direct access to a previously known entry; I think this property is enough to have lockless readers.